### PR TITLE
Add method to check if code is running in orchestrator

### PR DESCRIPTION
**Pull request by chmeliik**
* OSBS-7005

The method assumes that a build is an orchestrator build if (and only if) it includes the `orchestrate_build` buildstep plugin. It is defined in `DockerBuildWorkflow`, which seems like the proper place for it given that it contains most of the build configuration. Build plugins also get an `is_in_orchestrator()` method which simply calls the `is_orchestrator_build()` method of their workflow (for convenience).

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/inner.py [hunk 1]
```diff
@@ -448,6 +451,28 @@ def __init__(self, source, image, prebuild_plugins=None, prepublish_plugins=None
...
+        Get the orchestrate_build plugin configuration for this workflow
+        if present (will be present for orchestrator, not for worker).
+
+        :return: orchestrate_build plugin configuration dict or None
+        """
+        for plugin in self.buildstep_plugins_conf or []:
+            if plugin['name'] == PLUGIN_BUILD_ORCHESTRATE_KEY:
+                return plugin
+        # Not an orchestrator build
+        return None
```
Comment (twaugh) [inline atomic_reactor/inner.py]: Would it be more Pythonic for this to raise KeyError or something?

Comment (chmeliik) [reply atomic_reactor/inner.py]: KeyError felt a bit weird, I made it a ValueError, see if you like it :-) .

Comment (twaugh) [reply atomic_reactor/inner.py]: Looks great, thanks.

**Comment by MartinBasti** [comment]
Release notes:
-  method `is_in_orchestrator()` has been added to `Plugin` class as a common way how to determine whether code is running in orchestrator or worker.

**Pull request merged**